src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/Tokenizer.java

```diff
@@ -1,5 +1,8 @@
 package org.rstudio.studio.client.workbench.views.source.editors.text.ace;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+
 import com.google.gwt.core.client.JavaScriptObject;
 
 public class Tokenizer extends JavaScriptObject
```

Comment:
Does this code only run on reformat or does it run all the time? (affects how carefully I think we need to review it for correctness and edge cases?)